### PR TITLE
Don't error if autosave runs and there are no changes to save

### PR DESCRIPTION
**Pull request by notnownikki**
## Description

If autosave runs and there are no changes, the user gets an error banner. This happens when there's an autosave newer than the post content you're editing, and autosave runs again.

errorerror

This change sets the status code of the error to 200, because the fact that there are no changes since the last autosave shouldn't be an error we alert users to.

## Types of changes
Bug fix

### CODE REVIEW

File: lib/class-wp-rest-autosaves-controller.php [hunk 1]
```diff
@@ -309,7 +309,7 @@ public function create_post_autosave( $post_data ) {
 
 			if ( ! $autosave_is_different ) {
 				wp_delete_post_revision( $old_autosave->ID );
-				return new WP_Error( 'rest_autosave_no_changes', __( 'There is nothing to save. The autosave and the post content are the same.', 'gutenberg' ) );
+				return new WP_Error( 'rest_autosave_no_changes', __( 'There is nothing to save. The autosave and the post content are the same.', 'gutenberg' ), array( 'status' => 200 ) );
```
Comment (danielbachhuber) [inline lib/class-wp-rest-autosaves-controller.php]: I think `400` is more appropriate here than `200`. We should return `400` from the REST API, and then have special handling in Gutenberg for a `400` from the autosaves endpoint.

**Review by danielbachhuber — CHANGES_REQUESTED** [review]

**Review by danielbachhuber — APPROVED** [review]

File: editor/store/effects.js [hunk 2]
```diff
@@ -230,7 +230,14 @@ export default {
 		}
 	},
 	REQUEST_POST_UPDATE_FAILURE( action, store ) {
-		const { post, edits } = action;
+		const { post, edits, error } = action;
+
+		if ( error && 'rest_autosave_no_changes' === error.code ) {
```
Comment (aduth) [inline editor/store/effects.js]: In what situations will we expect `error` to be empty such that we'd want to check its truthiness? If it's expected, we should have a separate unit test for this circumstance as well.

**Review by aduth — APPROVED** [review]
>If autosave runs and there are no changes, the user gets an error banner.

Ideally, we should never be sending an autosave request in the first place if there are no changes to be saved. As currently implemented, I could imagine there may be cases where the user makes a change and manually resets it back to the original value where autosave could happen (particularly for post content). And regardless it does seem reasonable we don't want to surface/alarm the user on this type of error.

Would this apply for any autosave error (i.e. never show warnings on failed autosave)? I'd guess depending on the error type, we would want to show a notice, so fine as-is.

File: editor/store/effects.js (hunk 2 above)
Comment (notnownikki) [reply editor/store/effects.js]: I'm not sure here. While `REQUEST_POST_UPDATE` does include the error in the dispatched action, `REQUEST_POST_UPDATE_FAILURE` didn't use it until this change, and several tests didn't include an error either, so I felt it was safer to include this check until someone who knows things better could say for certain if the error would always be there or not.

**Pull request merged**